### PR TITLE
Plans (Storage): Introduce `addOnSlug` to add-ons. deprecate use of `featureSlugs` for storage

### PR DESCRIPTION
**Pull request by chriskmnds**
Related to #

## Proposed Changes

Follow-up to https://github.com/Automattic/wp-calypso/pull/92020 to introduce add-on slugs/keys to uniquely identify add-ons. This removes the need to include non-feature slugs in the `featureSlugs` property to reference the 50/100GB storage add-ons. The `featureSlugs` was initially put in place to reference plan/site features to be marked as "included with current site/plan" in the add-ons screen.

We can extend this refactor to re-index the add-ons structure to be of key-value form, although that's a bit of a heavier refactor.

## Why are these changes being made?

Important to preserve the right semantic identity of add-on properties / eliminate ambiguity. 

## Testing Instructions

* Ensure storage add-ons work as previously in signup - `/start/plans`
* Ensure current plan on Business/Entrepreneur in `/plans/[ site ]` is upgradeable with storage once (select to upgrade from the dropdown in spotlight card, then revisit `/plans/[ site ]` - it should show the upgraded volume along with additional cost)
* Ensure storage is automatically added to the cart when in query param e.g. `/start/business/?storage=50gb-storage-add-on`
* Ensure `/add-ons/[ site ]` renders all the add-ons correctly. Purchase an add-on and confirm it is marked as "purchased"

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by southp** [comment]
I haven't had a chance to look into the code yet, but wanted to drop a quick comment that it works as described :) 

* Following the test plan, everything works as expected ✅
     * I did it for several rounds, for different combinations of Creator/Entrepreneur * buying from the /add-on or /plans * different dropdowns in the spotlight card, features grid, comparison grid ✅
* Purchase a storage add-on on a Free site from /add-on ✅
* On a Creator site, refund the add-on, and make sure this change is reflected in /plans ✅

File: client/my-sites/plans-features-main/hooks/use-generate-action-callback.ts [hunk 1]
```diff
@@ -78,7 +78,8 @@ function useUpgradeHandler( {
 					product_slug: selectedStorageAddOn.productSlug,
 					quantity: selectedStorageAddOn.quantity,
 					volume: 1,
-					extra: { feature_slug: selectedStorageAddOn?.featureSlugs?.[ 0 ] },
+					// TODO: `feature_slug` below should be refactored to `add_on_slug` now that the two concepts are separated
+					extra: { feature_slug: selectedStorageAddOn?.addOnSlug },
```
Comment (chriskmnds) [inline client/my-sites/plans-features-main/hooks/use-generate-action-callback.ts:82]: We need to investigate the use of `feature_slug` here and how to update to `add_on_slug`. There are other similar instances in the code that will need to be accounted for.

Comment (chriskmnds) [reply client/my-sites/plans-features-main/hooks/use-generate-action-callback.ts:82]: Issue https://github.com/Automattic/wp-calypso/issues/92373

**Comment by chriskmnds** [comment]
Thanks for testing :-)

**Pull request merged**